source/darwin/darwin_pki_utils.c

```diff
@@ -22,7 +22,8 @@ int aws_import_public_and_private_keys_to_identity(
     CFAllocatorRef cf_alloc,
     const struct aws_byte_cursor *public_cert_chain,
     const struct aws_byte_cursor *private_key,
-    CFArrayRef *identity) {
+    CFArrayRef *identity,
+    const char *keychain_path) {
```

Comment:
this should be a byte_cursor if you don't need to preserve the null terminator or aws_string if you do.

Reply:
It's passed on to [SecKeychainOpen](https://developer.apple.com/documentation/security/1396431-seckeychainopen?language=objc) which appears to to expect a null terminated string buffer, so aws_string sounds like the way to go.